### ISSUE TITLE
input.system broken on OpenBSD

### ISSUE THREAD

**Issue by landryb** [influxdata/telegraf#5157]
Ported telegraf 1.9 to OpenBSD-current, using go 1.11.4. Trying the `inputs.system` plugin triggers an EINVAL somewhere, probably in gopsutils. Other input plugins (processes, mem, etc..) seem to work fine.

```
$cat /etc/telegraf/crash.conf                                                
[[outputs.file]]
   files = ["/tmp/metrics.out"]
[[inputs.mem]]
[[inputs.system]]

$telegraf -test -config /etc/telegraf/crash.conf  
2018-12-18T18:23:41Z I! Starting Telegraf 
> mem,host=c64.proxmox2 active=155381760i,available=3245314048i,available_percent=78.24602604010302,buffered=614449152i,cached=0i,commit_limit=0i,committed_as=0i,dirty=0i,free=3176251392i,high_free=0i,high_total=0i,huge_page_size=0i,huge_pages_free=0i,huge_pages_total=0i,inactive=69062656i,low_free=0i,low_total=0i,mapped=0i,page_tables=0i,shared=0i,slab=0i,swap_cached=0i,swap_free=0i,swap_total=0i,total=4147576832i,used=902262784i,used_percent=21.753973959896978,vmalloc_chunk=0i,vmalloc_total=0i,vmalloc_used=0i,wired=4096i,write_back=0i,write_back_tmp=0i 1545157421000000000
2018-12-18T18:23:41Z E! [telegraf] Error running agent: invalid argument
```

Looking for hints at how to debug this go thing, as egdb isnt really helpful, and i doubt delve is portable to OpenBSD (nor would anybody try to port it there)

**Comment by landryb** [comment]
Oh sorry, that's OpenBSD/amd64. prtinf-debugging is a bit meh.. and i dont really understand why, but ktracing should show syscalls and sysctls called, and none jumped straight out as the one failing so far.

**Comment by pkotsiop** [comment]
I've been playing with telegraf in openbsd the last 2 days. The problem is that telegraf seems to not get the latest version of golang.org/x/sys and thus misses a change in the perl script which creates the sysctl calls list, which now includes a new header file added (possibly on obsd 6.3) and holds the vm.* defines. I checked Gopkg.toml and seems to get the x/sys from the master branch, but I have no idea how GO works and maybe the "master" branch is not the "master" branch published on github repo of x/sys. 

PS: In case you need it for troubleshooting, the header file is "uvm/uvmexp.h"

**Comment by landryb** [comment]
Ah, nice debugging. Well, then i need to try using the sys package used is the one from the system (ie go-sys from openbsd pkgs), and not the random version bundled/packed by dep... i had put a comment about it in the port Makefile (https://github.com/openbsd/ports/blob/master/sysutils/telegraf/Makefile#L20), seems i need to revisit that.

**Comment by pkotsiop** [comment]
There is another problem with the cpu states reported, but it's not due to telegraf code, gopsutil needs updating. Just linking it here too, in case someone stumbles upon this issue.
https://github.com/shirou/gopsutil/issues/621

PS: I am also linking this here in case it helps landryb make a fully functional port quicker!! :p

**Comment by landryb** [comment]
sadly, same thing/crash/`Invalid argument` message if using go-sys, go-net, go-text and go-crypto from the versions installed by the OpenBSD packages.

**Comment by pkotsiop** [comment]
I make it work through telegraf repo. Steps follow:

1. go get github.com/influxdata/telegraf
2. gmake (to have telegraf download its deps)
3. Edit `vendor/golang.org/x/sys/unix/mksysctl_openbsd.pl` and add `uvm/uvmexp.h` to @headers
4. cd vendor/golang.org/x/sys/unix/
5. GOOS=openbsd GOARCH=amd64 ./mksysctl_openbsd.pl > zsysctl_openbsd_amd64.go
6. cd ../../../../../
7. Edit Makefile and comment line 31 (the one which gets the deps)
8. gmake
9. doas gmake install

This should do the work.

**Comment by landryb** [comment]
Apparently this was fixed at some point in x/sys (cf golang/sys@219bb53033fb8511ee74a8375a3bc672a727bdce) so Gopkg.lock might just need to be updated to point to a newer commit ? will test playing with that..

**Comment by landryb** [comment]
Yep, with an updated go-sys from packages, it works:
```
$/usr/obj/ports/telegraf-1.9.0/go/bin/telegraf  -test -config /etc/telegraf/crash.conf 
2018-12-20T17:12:00Z I! Starting Telegraf 
> system,host=c64.proxmox2 load1=1.45361328125,load15=0.4091796875,load5=0.865234375,n_cpus=2i,n_users=4i 1545325920000000000
> system,host=c64.proxmox2 uptime=1021916i 1545325920000000000
> system,host=c64.proxmox2 uptime_format="11 days, 19:51" 1545325920000000000
```

so to resolve the telegraf issue, one should bump the dependency to a newer x/sys commit.
